### ISSUE TITLE
Lab 03

### ISSUE THREAD

**Issue by castower** [DS4PS/cpp-527-spr-2020#6]
Hello all, I'm currently working on my Github page and I'm having difficulty getting the front page details to change. The header and footer details are changing, but the title of the first page is still 'My website' and the description is: 'This is where I will tell my friends way too much about me'. Am I getting ahead of myself with altering these details or should they have synced with the header and footer?

Here is the website: https://castower.github.io/

Thanks!
Courtney

**Comment by lecy** [comment]
The _confit.yml page controls global site options. As much as possible the page options are pushed to the pages themselves. It can sometimes be hard to sort out which part of the page is embedded in the page layout, and which part is included in the MD document. 

Check your index file, which is homepage (websites automatically look for "index.htm" when loading, so it is always the first page that loads). 

https://github.com/castower/castower.github.io/blob/master/index.html

```
---
layout: page
title: My website
subtitle: This is where I will tell my friends way too much about me
use-site-title: true
---
```
Note that this is often "index.md" in the GitHub context. All MD files are converted to HTML files behind the scenes once GitHub pages are activated (**markdown** is actually a derivation of HTML which stands for a **markup** language - markdown is a "dumbed-down version of markup", thus markdown).

**Comment by castower** [comment]
@lecy thanks so much! That fixed it!

**Comment by sunaynagoel** [comment]
@lecy Is there a way to insert a picture from my computer to the change the avatar?

**Comment by castower** [comment]
@sunaynagoel you can upload an image to the img folder under your
repository and then change the file path to the new image.

I also went under the CSS folder and altered the main file to remove the
background-radius element so that it wasn't circular. I think it's around
line 170 or so.

EDIT: I went back and checked, it's border-radius on lines 177 and 186.

**Comment by sunaynagoel** [comment]
> @sunaynagoel you can upload an image to the img folder under your repository and then change the file path to the new image. I also went under the CSS folder and altered the main file to remove the background-radius element so that it wasn't circular. I think it's around line 170 or so.
> […](#)

Thanks Courtney

**Comment by sunaynagoel** [comment]
@lecy I noticed that there are buttons for a lot of social media like facebook, youtube and instagram etc. I wanted to bring in pinterest button and I could not. Is there way out ?

**Comment by lecy** [comment]
@sunaynagoel You would have to define a new page element first (you can try, but not at all required): 

https://github.com/castower/castower.github.io/blob/master/_includes/social-share.html

https://fontawesome.com/icons/pinterest?style=brands

**Comment by jmacost5** [comment]
I am really not understanding what I am doing on this project I have literally went in circles for the past 3 hours trying to figure it out alone. I can not get the file to copy and I am not understanding how.

**Comment by lecy** [comment]
@jmacost5 We need more information to be helpful. Where are you at in the steps? Which file are you trying to copy?

**Comment by sunaynagoel** [comment]
@lecy If I wanted to change the link of the news article on home page, how would I do that ?

**Comment by sunaynagoel** [comment]
> @sunaynagoel You would have to define a new page element first (you can try, but not at all required):
> 
> https://github.com/castower/castower.github.io/blob/master/_includes/social-share.html
> 
> https://fontawesome.com/icons/pinterest?style=brands

I could not make it to work at this point. But I know now what to do. I was able incorporate a link of the Pinterest account for the time being .

**Comment by lecy** [comment]
A blog?

**Comment by lecy** [comment]
@sunaynagoel What do you mean by news article on home page?

**Comment by sunaynagoel** [comment]
> @sunaynagoel What do you mean by news article on home page?

There are five article shown in main page ... are they blog post ? how can I change it. 

Here is the link for my page. I wanted to change those posts to be relevant to my website. https://sunaynagoel.github.io

**Comment by lecy** [comment]
Yep, they are just blog posts. Search for Jekyll blogs if you want more info. They are located in the posts folder. 

https://github.com/castower/castower.github.io/tree/master/_posts

**Comment by jmacost5** [comment]
> @jmacost5 We need more information to be helpful. Where are you at in the steps? Which file are you trying to copy?

I guess my first question is are we supposed to be coping the page from beautiful-jekyll by messing with our repositories that we made, I am not understanding how

**Comment by lecy** [comment]
Yep. Once you have forked the repo all of those files are yours. Go to the layouts folder and create a new file, name it “liquid-tables.html” and copy content from Barebones Jekyll site. That will create a new HTML page template for you. Then you can use the template with new pages by changing the layout to liquid-table in the YAML header of new pages.

Goal is to start with a site that works, and figure out how you can customize - change page name, add a new page, etc.



This message was sent from a mobile device.

**Comment by jmacost5** [comment]
> Yep. Once you have forked the repo all of those files are yours. Go to the layouts folder and create a new file, name it “liquid-tables.html” and copy content from Barebones Jekyll site. That will create a new HTML page template for you. Then you can use the template with new pages by changing the layout to liquid-table in the YAML header of new pages. Goal is to start with a site that works, and figure out how you can customize - change page name, add a new page, etc. This message was sent from a mobile device.
> […](#)
> On Feb 6, 2020, at 7:28 AM, jmacost5 ***@***.***> wrote: @jmacost5 We need more information to be helpful. Where are you at in the steps? Which file are you trying to copy? I guess my first question is are we supposed to be coping the page from beautiful-jekyll by messing with our repositories that we made, I am not understanding how — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

I think my website is not loading properly...
https://jmacost5.github.io/jmacost5lab3/

**Comment by castower** [comment]
@jmacost5 I was also confused at first. You can skip the step about creating your own repository and just follow the instructions under the readme for beautiful-jekyll. There's a gif that you can follow and it shows you step-by-step. I just opened two tabs, one with the instructions and another to use to fork the repository. You'll name it after forking.

**Comment by sunaynagoel** [comment]
> Yep, they are just blog posts. Search for Jekyll blogs if you want more info. They are located in the posts folder.
> 
> https://github.com/castower/castower.github.io/tree/master/_posts

Thank you. This was helpful

**Comment by sunaynagoel** [comment]
> > Yep. Once you have forked the repo all of those files are yours. Go to the layouts folder and create a new file, name it “liquid-tables.html” and copy content from Barebones Jekyll site. That will create a new HTML page template for you. Then you can use the template with new pages by changing the layout to liquid-table in the YAML header of new pages. Goal is to start with a site that works, and figure out how you can customize - change page name, add a new page, etc. This message was sent from a mobile device.
> > […](#)
> > On Feb 6, 2020, at 7:28 AM, jmacost5 _**@**_.***> wrote: @jmacost5 We need more information to be helpful. Where are you at in the steps? Which file are you trying to copy? I guess my first question is are we supposed to be coping the page from beautiful-jekyll by messing with our repositories that we made, I am not understanding how — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.
> 
> I think my website is not loading properly...
> https://jmacost5.github.io/jmacost5lab3/

@jmacost5 mine loaded exactly like this in the beginning. I kept on following the steps and it worked out. Hope this helps.

**Comment by jmacost5** [comment]
I just got the following error in my email, is this going to cause me to not get the website to work right?

The page build failed for the `master` branch with the following error:

You have an error on line 25 of your `_config.yml` file. For more information, see https://help.github.com/en/github/working-with-github-pages/troubleshooting-jekyll-build-errors-for-github-pages-sites#config-file-error.

For information on troubleshooting Jekyll see:

  https://help.github.com/articles/troubleshooting-jekyll-builds

If you have any questions you can contact us by replying to this email

https://jmacost5.github.io/


I am also confused that my website is not updating with the changes that I made.

**Comment by Jigarci3** [comment]
Hi @jmacost5 , I also received an error when I edited my code. It was helpful to look back on the original code to figure out what went wrong with the changes I had made. I also noticed that my website would not show any of the changes I was making when there was an error but updated as soon as that error was fixed!

**Comment by lecy** [comment]
Hi All -

I was on the road to present some research and caught a bug on the plane, so I am just getting back to these questions today. Apologies for the slow response time. 

Thank you to all who provided helpful responses! 

@jmacost5 Regarding the 404 page you received: 

https://jmacost5.github.io/jmacost5lab3/

It was because of you added "/jmacost5lab3/" to the end of the URL. I see that the main site is operational:  

https://jmacost5.github.io/

For your build error, can you please send a link to your config file:

https://github.com/DS4PS/barebones-jekyll/blob/master/_config.yml

As @Jigarci3 noted, the difference view can be helpful for examining changes that might have impacted your code. You can get to the history by clicking the 

![image](https://user-images.githubusercontent.com/1209099/74110712-1f1fd900-4b4c-11ea-8f78-6e861b1a82f4.png)

It will show you which changes were made at specific lines:

![image](https://user-images.githubusercontent.com/1209099/74110745-4aa2c380-4b4c-11ea-8355-b8f159928d7c.png)

**Comment by TVK36692** [comment]
I literally just changed the first section and i get an error that the page build failed on line 4:

# --- General options --- #

# Name of website
title: My First GitHub

# Short description of your site
description: A site to start using GitHub on a regular basis.

I just changed the title and description and it already fails.

Am i doing something wrong here?

**Comment by TVK36692** [comment]
weird that it works here.. "# --- General options --- #

# Name of website
title: My First GitHub

# Short description of your site
description: A site to start using GitHub on a regular basis."

**Comment by castower** [comment]
@TVK36692 I got the same error so I put my title and subtitle text in quotation marks (" ") and that fixed it. I think it's picky if there's an extra space somewhere.

**Comment by JaesaR** [comment]
My table demo is not working. Under nav-bar links I have the following: 
 Table Demo: "table-demo"

I also have downloaded the liquid-tables.html to layouts and have uploaded a table-demo.md file to my repository. I also edited the table-demo file to "layout: default"

However, I keep getting a 404 error saying that the table-demo page does not exist. How can I fix this?

**Comment by lecy** [comment]
@TVK36692 I would second what Courtney said - YAML fields are sensitive both to the number of spaces you use (if they are hierarchical" as well as to some special characters and spaces. So it's good practice to put phrases in quote marks, and avoid colons. 

Please include a link to your file on GitHub if you want us to help problem solve.

**Comment by lecy** [comment]
@JaesaR 

In your header your are asking to use the liquid-table page layout: 

````
---
layout: liquid-table
---
````
Try changing the layout file "Liquid-table.html" to "liquid-table.html", or change the layout in the header to "Liquid-table".

**Comment by JaesaR** [comment]
@lecy I changed everything to make sure the capital L was accounted for and it is still giving me the same 404 error.

Also, I keep getting emails saying "You have an error on line 23 of your `_config.yml` file."  Which is weird because line 23 is "About Me: 'aboutme'" and my About Me is working fine..?

**Comment by lecy** [comment]
The error prevents the website from updating, so it will still show the last pages that were able to render. 

I think you have an extra space in your navbar before Table Demo. That navbar list starts at line 23 and ends with Table Demo. 

```r
navbar-links:
  About Me: "aboutme"
  Resources:
    - Beautiful Jekyll: "http://deanattali.com/beautiful-jekyll/"
    - Learn markdown: "http://www.markdowntutorial.com/"
    - Getting Started: "gettingstarted"
   Table Demo: "table-demo"
```

Otherwise, maybe the punctuation in the YAML field on your home (index.html) page? Special characters can break parsers sometimes (not sure how the YAML header is parsed so this is just a guess). Remove punctuation or enclose the sentence with quotes.  

```
subtitle: If you googled my name and ended up here, hi! This website is for a class.
```
And maybe the site name too? 

```
title: Jaesa's Website
```

**Comment by lecy** [comment]
@JaesaR It looks like you changed *table-demo.md" to "table-demo" and "gettingstarted.md" to "gettingstarted". 

They need the .md file extensions so that GitHub pages will know to convert them to HTML pages on the server-side. Otherwise you get a weird behavior where you get a file download option when you use your navigation bar.

**Comment by cjbecerr** [comment]
I'm not seeing week 5 practice problem and lab 04 open, have these been made available for others? @lecy

**Comment by lecy** [comment]
The open policing data I had planned on using for the lab did not work out, unfortunately, so I had to create a new lab from scratch. I am going to post the lab and an updated schedule for the remaining work later today.
